data/spells/scripts/support/findPerson.lua

```diff
@@ -0,0 +1,133 @@
+local LEVEL_LOWER = 1
+local LEVEL_SAME = 2
+local LEVEL_HIGHER = 3
+
+local DISTANCE_BESIDE = 1
+local DISTANCE_CLOSE = 2
+local DISTANCE_FAR = 3
+local DISTANCE_VERYFAR = 4
+
+function onCastSpell(creature, var)
+	if not creature:getPlayer() then
+		return false
+	end
+
+	local targetPlayer = Player(Variant.getString(var))
+	if not targetPlayer then
+		return false
+	end
+
+	if targetPlayer:getGroup():getAccess() and not creature:getGroup():getAccess() then
+		creature:sendCancelMessage(RETURNVALUE_PLAYERWITHTHISNAMEISNOTONLINE)
+		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local playerPosition = creature:getPosition()
+	local targetPosition = targetPlayer:getPosition()
+	local distance = {
+		["x"] = playerPosition.x - targetPosition.x,
+		["y"] = playerPosition.y - targetPosition.y,
+		["z"] = playerPosition.z - targetPosition.z
+	}
```

Comment:
Yes but it looks cleaner if you add it to an array, also less things to write out.